app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user 
+      session[:user_id] = user.id
+      redirect_to posts_path, notice: "Logged in!"
+    else
+      flash.now[:alert] = "Email or password is invalid"
```

Comment:
As, You have mentioned password invalid message, Where are you checking password ?